src/Poseidon/Package.hs

```diff
@@ -285,6 +285,17 @@ readPoseidonPackageCollection opts baseDirs = do
             if _readOptOnlyLatest opts
             then filterM (isLatestInCollection loadedPackages) loadedPackages
             else pure $ loadedPackages
+    -- check if any packages appear more than once
```

Comment:
Why did you introduce this check? I mean, I get it, you want to avoid having _exactly_ the same package (and version) present, but it seems a rather unlikely case for me to cater for. I mean... why not, but I don't think it's related to the new changes we discussed last, right?

Reply:
Well - it's an important constraint, because loading copies of a package can have unforeseen consequences. And I can easily imagine a situation where a user accidentally copies a package and then runs into this situation.

And I think this is just a more general implementation of the rule you suggested yesterday for `isLatestInCollection`...

Reply:
OK I agree